powertools-parameters/src/test/java/software/amazon/lambda/powertools/parameters/ParamManagerIntegrationTest.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates.
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package software.amazon.lambda.powertools.parameters;
+
+import org.assertj.core.data.MapEntry;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import software.amazon.awssdk.services.secretsmanager.SecretsManagerClient;
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueRequest;
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueResponse;
+import software.amazon.awssdk.services.ssm.SsmClient;
+import software.amazon.awssdk.services.ssm.model.*;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+import static org.mockito.MockitoAnnotations.openMocks;
+
+public class ParamManagerIntegrationTest {
+
+    @Mock
+    SsmClient ssmClient;
+
+    @Captor
+    ArgumentCaptor<GetParameterRequest> ssmParamCaptor;
+
+    @Captor
+    ArgumentCaptor<GetParametersByPathRequest> ssmParamByPathCaptor;
+
+    @Mock
+    SecretsManagerClient secretsManagerClient;
+
+    @Captor
+    ArgumentCaptor<GetSecretValueRequest> secretsCaptor;
+
+
+    @BeforeEach
+    public void setup() throws NoSuchFieldException, IllegalAccessException {
+        openMocks(this);
+
+        // A bit dirty but necessary as everything is static in ParamManager
+        Field ssmProviderField = ParamManager.class.getDeclaredField("ssmProvider");
```

Comment:
We have used commons lang  in test context in other module to use Apache reflection utils. If needed you might check that too.